src/Microsoft.Azure.WebJobs.Host/JobHostConfiguration.cs

```diff
@@ -164,6 +167,17 @@ public string StorageConnectionString
         /// </summary>
         public TimeSpan? FunctionTimeout { get; set; }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether function invocations will timeout when
+        /// a <see cref="FunctionTimeout"/> is specified and a debugger is attached. False by default.
+        /// </summary>
+        public bool TimeoutWhileDebugging { get; set; }
+
+        /// <summary>
+        /// When true, an exception is thrown when a function timeout expires.
+        /// </summary>
+        public bool ThrowExceptionOnFunctionTimeout { get; set; }
```

Comment:
Check out my latest commit. I've given it some quick testing from Script and looks good. I pushed the grace period into the core SDK and made it configurable.

---
In reply to: [76665848](https://github.com/Azure/azure-webjobs-sdk/pull/799#discussion_r76665848) [](ancestors = 76665848)